epictrack-web/src/components/layout/SideNav/SideNav.tsx

```diff
@@ -72,9 +74,36 @@ const DrawerBox = () => {
     return <Icon className={`sidebar-item ${active ? "active" : ""}`} />;
   }, []);
 
-  const groupedRoutes = React.useMemo(
-    () => groupBy(Routes, (p) => p.group),
-    []
+  const isRouteAllowed = (route: RouteType, roles: Array<any>) => {
+    const allowed = route["allowedRoles"] ?? [];
+    return hasPermission({ roles, allowed }) || !route["isAuthenticated"];
+  };
+
+  const filterRoutes = (routes: RouteType[], roles: Array<any>) => {
+    return routes.reduce((allowedRoutes: any, route) => {
+      if (isRouteAllowed(route, roles)) {
+        const newRoute: RouteType = { ...route };
+        if (newRoute["routes"]) {
+          newRoute["routes"] = filterRoutes(newRoute["routes"], roles);
+        }
+
+        const addRoute =
+          (newRoute["routes"] && newRoute["routes"].length > 0) ||

```

Comment:
do we need this? can't we just do allowedRoutes.push(newRoute) in the if statement above it

Reply:
No because itll only add the navbar elements that have nested elements because of that if (newRoute["routes"])